mewarpx/mewarpx/mwxrun.py

```diff
@@ -103,27 +103,40 @@ def init_timestep(self, V_anode, DT=None, CFL_factor=None, V_grid=5):
         return self.dt
 
 
-    def init_run(self):
+    def init_run(self, restart=None, checkpoint_prefix="chkpoint", additional_steps=None):
         if self.initialized:
             raise RuntimeError(
                 "Attempted to initialize the mwxrun class multiple times.")
-
-        self.simulation.initialize_inputs()
-        self.simulation.initialize_warpx()
-
-        self.me = _libwarpx.libwarpx.warpx_getMyProc()
-        self.n_procs = _libwarpx.libwarpx.warpx_getNProcs()
-
-        # loop over all species to build their pid_dict's
-        for species in self.simulation.species:
-            species.init_pid_dict()
-
-        # A level is needed for many things like level number. For now I'm
-        # statically setting the default level here. I'm not sure of pitfalls
-        # or how to handle it more generally yet.
-        self.lev = _libwarpx.libwarpx.warpx_finestLevel()
-
-        self.initialized = True
+        try:
+            if restart != False:
+                init_restart_util.run_restart(checkpoint_prefix, restart is not None, additional_steps)
```

Comment:
Why do `restart is not None` instead of just `restart`?

Reply:
That was in order to not have that parameter have possible values of `True` or `None`, because that seems strange, so I converted it to a boolean.